lib/commands/exam.js

```diff
@@ -332,43 +332,45 @@ module.exports = TestCommand.extend({
       // If either timers is set, we should record the failed browser and call browserExitHandler
       if (this.finished && (!this.onProcessExitTimer && !this.pendingTimer)) {
         return;
-      } else if (commands.get('writeExecutionFile')) {
+      }
+      if (commands.get('writeExecutionFile')) {
+        // when a browser fails with an error record failed browser id.
         const browserId = getBrowserId(this.launcher.settings.test_page);
         testemEvents.recordFailedBrowserId(browserId);
       }
 
       browserExitHandler.call(this, true);
     };
 
-    return this._getModuleMetadataAndBrowserExitSocketEvents(browserExitHandler, browserFailureHandler);
+    return this._getModuleMetadataAndBrowserExitSocketEvents(browserExitHandler, browserTerminationHandler);
   },
 
   /**
    * Add browser socket events are needed for both with load-balance and without load-balance
    *
    * @param {Object} browserExitHandler
-   * @param {Object} browserFailureHandler
+   * @param {Object} browserTerminationHandler
    */
-  _getModuleMetadataAndBrowserExitSocketEvents(browserExitHandler, browserFailureHandler) {
+  _getModuleMetadataAndBrowserExitSocketEvents(browserExitHandler, browserTerminationHandler) {
     const events = {};
     let init = false;
 
     events['tests-start'] = function() {
       if (!init) {
-        this.process.on('processExit', browserFailureHandler.bind(this));
-        this.process.on('processError', browserFailureHandler.bind(this));
+        // when test runs without launcher (with --no-launch) process object is not avaialble.
+        if (typeof this.process === 'undefined' || this.process === null) {
```

Comment:
@choheekim I remember when you were debugging an issue, this line of code was of interest and may have been a contributor to the issue. What was the outcome of the investigation again?

Reply:
The line was an issue in `disconnect` event where a browser is disconnect and a process is still up a browser is not able to fall under if statement to execute `browserFailureHandler`. 

However, verifying whether process is available in `tests-start` is not required as when a browser is launched manually process is not available then registering callback handler for tests-start will be failing - https://github.com/ember-cli/ember-exam/issues/378

Reply:
@choheekim is `tests-start is not required` a typo?

Reply:
Woops, it was a typo. Edit the comment :)